nnstreamer/tensor_trainer/tensor_trainer_nntrainer.cc

```diff
@@ -260,24 +267,9 @@ void NNTrainer::InputTensorsInfo::getSample(float **input, float **label,
     pop_count = 0;
   }
 
-  if (is_data_full_locked && queue_count > 0) {
-    data_full.notify_one();
-    ml_logd("send signal");
-  }
+  int queue_count = (queue_rear - queue_front + queue_size) % queue_size;
   ml_logd("front:%d, rear:%d, filled:%d", queue_front, queue_rear, queue_count);
-
-  /* epoch is complete */
-  if (pop_count == 0)
-    return;
-
-  /* to avoid dead lock, check is_data_full_locked */
-  if (!is_data_full_locked && queue_count == 0) {
-    ml_logd("locked, need to wait for more data");
-    std::unique_lock<std::mutex> lock(data_wait_lock);
-    is_data_wait_locked = TRUE;
-    data_wait.wait(lock);
-    ml_logd("unlocked, get data");
-  }
+  data_full.notify_one();
```

Comment:
```lock.unlock()``` missing before notify_one

Reply:
data_full's lock is in nntrainer_model_push_data(), if isQueueFull() is TRUE, locked
```
  std::unique_lock<std::mutex> lock(data->queue_lock);
  data->data_full.wait(lock, [&] { return !data->isQueueFull(); });
 ```
data_full.notify_one() is called ``` data->data_full.wait(lock, [&] { return !data->isQueueFull(); });``` is unlocked